core/dbt/context/base.py

```diff
@@ -556,6 +558,10 @@ def log(msg: str, info: bool = False) -> str:
               {{ log("Running some_macro: " ~ arg1 ~ ", " ~ arg2) }}
             {% endmacro %}"
         """
+        try:
+            msg = str(msg)
+        except Exception as e:
+            raise RuntimeException("log failed to stringify the given object") from e
```

Comment:
Did a one-off test for this, since  I can't think of a convenient way to induce this within the test case via jinja provided input (given dbt/jinja design, I'd sort of hope not 😆)


<img width="760" alt="image" src="https://user-images.githubusercontent.com/185076/185770359-8fcf12fa-80cd-431d-b464-2bc9bf913cbf.png">